src/App.jsx

```diff
@@ -1,107 +1,45 @@
-
-        <p className="PostInfo__body">
-          est rerum tempore vitae sequi sint nihil reprehenderit dolor beatae
-          ea dolores neque fugiat blanditiis voluptate porro vel nihil
-          molestiae ut reiciendis qui aperiam non debitis possimus qui neque
-          nisi nulla
-        </p>
+function getUserByPostId(id) {
+  const arr = [];
 
-        <hr />
+  commentsFromServer.filter((elem) => {
+    if (elem.postId === id) {
+      arr.push(elem);
+    }
 
-        <b data-cy="NoCommentsMessage">No comments yet</b>
-      </div>
+    return elem;
+  });
 
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">
-            doloremque illum aliquid sunt
-          </h3>
+  return arr;
+}
 
-          <p>
-            {' Posted by  '}
+function getUserByUserId(userId) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
 
-            <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-              Patricia Lebsack
-            </a>
-          </p>
-        </div>
+export const posts = postsFromServer.map(post => ({
+  ...post,
+  user: getUserByUserId(post.userId),
+}));
 
-        <p className="PostInfo__body">
-          deserunt eos nobis asperiores et hic est debitis repellat molestiae
-          optio nihil ratione ut eos beatae quibusdam distinctio maiores earum
-          voluptates et aut adipisci ea maiores voluptas maxime
-        </p>
+export const postsUpdate = posts.map((elem) => {
+  const valueElem = elem;
 
-        <div className="CommentList">
-          <div className="CommentInfo">
-            <div className="CommentInfo__title">
-              <strong className="CommentInfo__name">pariatur omnis in</strong>
+  valueElem.comments = getUserByPostId(elem.id);
 
-              {' by '}
+  return elem;
+});
 
-              <a
-                className="CommentInfo__email"
-                href="mailto:Telly_Lynch@karl.co.uk"
-              >
-                Telly_Lynch@karl.co.uk
-              </a>
-            </div>
-
-            <div className="CommentInfo__body">
-              dolorum voluptas laboriosam quisquam ab totam beatae et aut
-              aliquid optio assumenda voluptas velit itaque quidem voluptatem
-              tempore cupiditate in itaque sit molestiae minus dolores magni
-            </div>
-          </div>
-
-          <div className="CommentInfo">
-            <div className="CommentInfo__title">
-              <strong className="CommentInfo__name">
-                odio adipisci rerum aut animi
-              </strong>
-
-              {' by '}
-
-              <a
-                className="CommentInfo__email"
-                href="mailto:Nikita@garfield.biz"
-              >
-                Nikita@garfield.biz
-              </a>
-            </div>
+export const App = () => (
+  <section className="App">
+    <h1 className="App__title">Static list of posts</h1>
```

Comment:
Move or logic inside this component (postsUpdate, getUserByUserId ... ) and return this JSX you have here now. It is better if what is used only in this component is in the same scope and it also makes your code easier to debug, test and read.